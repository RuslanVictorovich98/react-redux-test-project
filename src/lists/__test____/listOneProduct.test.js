import React from 'react';
import {shallow} from 'enzyme';
import ListOneProduct from '../listOneProduct';

describe('TEST component listOneProduct', () => {
    const props = {
        product: undefined,
    }

    it('TEST render to component', () => {
        const nextProps = {
            ...props, 
            product: {img: ''}
        }
        const wrapper = shallow(<ListOneProduct {...nextProps} />)
        expect(wrapper.find('.content').text()).toMatchSnapshot();
    })

    it('ListOneProduct component', () => {
        const nextProps = {
            ...props, 
            product: {img: ''}
        }
        const wrapper = shallow(<ListOneProduct {...nextProps} />)
        expect(wrapper.exists()).toBe(true) // Returns whether or not any nodes exist in the wrapper.
    })

    it('Count element to component', () =>  {
        const nextProps = {
            ...props, 
            product: [{img: ''}]
        }
        const wrapper = shallow(<ListOneProduct {...nextProps} />)
        expect(wrapper.find('.content-body')).toHaveLength(1)
    })
})