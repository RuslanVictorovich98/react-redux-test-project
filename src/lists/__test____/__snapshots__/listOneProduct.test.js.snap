// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST component listOneProduct TEST render to component 1`] = `" Имя:   Бренд:   Категория:   Цена:   Звезд:   "`;
